apps/tailwind-components/components/input/Ref.vue

```diff
@@ -88,6 +88,31 @@ onMounted(async () => {
   initialCount.value = count.value;
 });
 
+// the selectionMap is not reactively bound to the model due to need to fetch the options asynchonously
+watch(
+  () => modelValue.value,
+  () => {
+    if (props.isArray === false) {
```

Comment:
this only fires on change right? 

I think we have another issue because the template might be using fields not in the modelValue (because that might be pkey) so applyTemplate is not enough. Sadly, we would need to retrieve the rows based on the primary key to ensure we can render. We really need to see if we can move the templates to server side because this is getting quite expensive.